R/logit_mle_newton.R

```diff
@@ -5,13 +5,35 @@ are_all_close <- function(v, w, abs_tol = 1e-6, rel_tol = 1e-6) {
   return(are_all_within_atol && are_all_within_rtol)
 }
 
-logit_newton = function(design, outcome, maxiter = 1000){
+take_one_newton_step = function(design, outcome, beta_old, newton_opt = "qr"){
```

Comment:
This is getting large enough I'd consider splitting it up. Either into one function per solver, or pulling out some of the qr code. Just like how the hess/grad have their own functions, you could have functions to compute x/y_adjusted